src/analysis/convex/basic.lean

```diff
@@ -332,6 +331,30 @@ begin
         ht (mem_prod.1 hx).2 (mem_prod.1 hy).2 ha hb hab⟩
 end
 
+lemma directed.convex_Union {ι : Sort*} {s : ι → set E} (hdir : directed has_subset.subset s)
```

Comment:
Can't you deduce the directed version from the directed_on version, or conversely, instead of duplicating the proof?

Reply:
Indeed I can!